xml/hwbp_registers.xml

```diff
@@ -79,19 +83,23 @@
 
             This bit is only writable from Debug Mode.
         </field>
-        <field name="data" bits="MXLEN-6:0" access="R/W" reset="Preset">
+        <field name="data" bits="XLEN-6:0" access="R/W" reset="Preset">
             Trigger-specific data.
         </field>
     </register>
 
     <register name="Trigger Data 2" short="tdata2" address="0x7a2">
         Trigger-specific data.
-        <field name="data" bits="MXLEN-1:0" access="R/W" reset="Preset" />
+
+        If XLEN is less than DXLEN, writes to this register are sign-extended.
+        <field name="data" bits="XLEN-1:0" access="R/W" reset="Preset" />
     </register>
 
     <register name="Trigger Data 3" short="tdata3" address="0x7a3">
         Trigger-specific data.
-        <field name="data" bits="MXLEN-1:0" access="R/W" reset="Preset" />
+
+        If XLEN is less than DXLEN, writes to this register are sign-extended.
```

Comment:
"writes to this register" so you mean if I am in M-mode, and MXLEN < DXLEN, then the M-mode write will be sign-extended? I'm not saying you need to add more text, just making sure I understand your intention.

Reply:
Yes. Maybe the paragraph at the beginning of the trigger section should mention that there are always DXLEN storage bits. This line is saying that when writing it when XLEN<DXLEN, all storage bits are in fact updated, and the high bits are determined by sign extending.